database/sql/V072__auction_solution_orders.sql

```diff
@@ -0,0 +1,66 @@
+-- All auctions ran by autopilot
+CREATE TABLE competition_auctions (
+   id bigint PRIMARY KEY,
```

Comment:
I think we are [supposed to](https://stackoverflow.com/a/69109131) use [identity columns](https://www.postgresql.org/docs/current/ddl-identity-columns.html#DDL-IDENTITY-COLUMNS) to have the DB automatically generate these unique values for us.
```suggestion
   id bigint PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
```

Reply:
I understand it looks neat to use DB generated ID, but why would we use lock us in with using it?
Besides, right now `auctions` and `competition_auctions` need to be aligned with ids.

Reply:
>  Besides, right now auctions and competition_auctions need to be aligned with ids.
Sorry, the comment was supposed to be on `proposed_solutions.id`.

> I understand it looks neat to use DB generated ID, but why would we use lock us in with using it?

For Ids that have no other purpose than being unique and identifying rows I think it makes the most sense to let the DB make sure that things are unique instead of relying on domain code that can have bugs in that regard. Since we don't expect any additional information in the ID any value is as good as any other value as long as it's unique so why should we bother with maintaining that uniqueness ourselves?

Reply:
> For Ids that have no other purpose than being unique and identifying rows

But this is not actually true in this case right? Auction id is read directly by client and used to fetch data from other tables etc. It's not like it's inserted only once and never used by client but only by database internally to join on other tables etc.

But anyway, in this case we have to go with client defined ids because of:

>  right now auctions and competition_auctions need to be aligned with ids.

Reply:
Sorry, formatted my response incorrectly so the important part was in the quoted text.
The whole discussion was supposed to be about the ID of the solution, not the auction_id so it still stands IMO.